packages/nimble-components/src/toggle-button/styles.ts

```diff
@@ -1,20 +1,34 @@
 import { css } from '@microsoft/fast-element';
-import { focusVisible } from '../utilities/style/focus';
 
 import {
-    borderHoverColor,
-    borderWidth,
+    buttonAccentOutlineFontColor,
+    buttonLabelFontColor,
     fillSelectedColor,
-    fillSelectedRgbPartialColor
+    fillSelectedRgbPartialColor,
+    iconColor
 } from '../theme-provider/design-tokens';
-import { styles as buttonStyles } from '../patterns/button/styles';
+import {
+    buttonAppearanceVariantStyles,
+    styles as buttonStyles
+} from '../patterns/button/styles';
+import { appearanceBehavior } from '../utilities/style/appearance';
+import { ButtonAppearance } from './types';
 
 export const styles = css`
+    @layer base, base-selected, hover, focusVisible, active, disabled, top;
```

Comment:
This kind of override is really hard for me to follow mentally. II don't actually know how redefining `@layer` works. Sucks that it's also hard to inspect in dev tools. If I manually serialize the adoptedStylesheet in storybook with:

```js
console.log(Array.from($0.shadowRoot?.adoptedStyleSheets ?? []).flatMap(sheet => Array.from(sheet.rules).flatMap(rules => rules.cssText.split(';').join(';\n'))).join('\n\n') || 'none');
```

I get:

```
@layer base, base-selected, hover, focusVisible, active, disabled, top;


@layer base, hover, focusVisible, active, disabled, top;


@layer base {
  :host([hidden]) { display: none;
 }
  :host { display: inline-flex;
 }
  :host { background-color: transparent;
 height: var(--ni-nimble-control-height);
 color: var(--ni-nimble-button-label-font-color);
 font: var(--ni-nimble-button-label-font);
 cursor: pointer;
 outline: none;
 border: none;
 box-sizing: border-box;
 vertical-align: middle;
 }
  .control { background-color: transparent;
 height: 100%;
 width: 100%;
 border: var(--ni-nimble-border-width) solid transparent;
 box-sizing: border-box;
 color: inherit;
 border-radius: inherit;
 fill: inherit;
 display: inline-flex;
 align-items: center;
 justify-content: center;
 gap: 4px;
 cursor: inherit;
 font: inherit;
 outline: none;
 margin: 0px;
 padding: 0 var(--ni-nimble-standard-padding);
 position: relative;
 transition: box-shadow var(--ni-nimble-small-delay) ease-in-out,
                border-color var(--ni-nimble-small-delay) ease-in-out,
                background-size var(--ni-nimble-small-delay) ease-in-out;
 background-size: 100% 100%;
 background-repeat: no-repeat;
 background-position: center center;
 }
  :host([content-hidden]) .control { aspect-ratio: 1 / 1;
 padding: 0px;
 }
  .control::before { content: "";
 position: absolute;
 width: 100%;
 height: 100%;
 pointer-events: none;
 box-sizing: border-box;
 outline: transparent solid 0px;
 color: transparent;
 background-clip: content-box;
 transition: outline var(--ni-nimble-small-delay) ease-in-out;
 }
  .content { display: contents;
 }
  :host([content-hidden]) .content { display: inline-block;
 height: 1px;
 width: 1px;
 position: absolute;
 margin: -1px;
 clip: rect(1px, 1px, 1px, 1px);
 clip-path: inset(50%);
 overflow: hidden;
 padding: 0px;
 }
  [part="start"], [part="end"] { display: contents;
 --ni-nimble-icon-color: var(--ni-nimble-button-label-font-color);
 }
}

@layer hover {
  .control:hover { border-color: var(--ni-nimble-border-hover-color);
 box-shadow: 0px 0px 0px var(--ni-nimble-border-width) var(--ni-nimble-border-hover-color) inset;
 }
}

@layer focusVisible {
  .control:focus-visible { border-color: var(--ni-nimble-border-hover-color);
 box-shadow: 0px 0px 0px var(--ni-nimble-border-width) var(--ni-nimble-border-hover-color) inset;
 }
  .control:focus-visible::before { outline: var(--ni-nimble-border-width) solid var(--ni-nimble-border-hover-color);
 outline-offset: -3px;
 }
}

@layer active {
  .control:active { box-shadow: none;
 color: var(--ni-nimble-button-label-font-color);
 border-color: var(--ni-nimble-border-hover-color);
 background-image: linear-gradient(
                var(--ni-nimble-fill-selected-color),
                var(--ni-nimble-fill-selected-color)
            );
 background-size: calc(100% - 2px) calc(100% - 2px);
 }
  .control:active [part="start"], .control:active [part="end"] { --ni-nimble-icon-color: var(--ni-nimble-button-label-font-color);
 }
  .control:active::before { outline: none;
 }
}

@layer disabled {
  :host([disabled]) { cursor: default;
 }
  :host([disabled]) .control { color: var(--ni-nimble-button-label-disabled-font-color);
 box-shadow: none;
 background-image: none;
 background-size: 100% 100%;
 }
  :host([disabled]) .control::before { box-shadow: none;
 }
  :host([disabled]) slot[name="start"]::slotted(*), :host([disabled]) slot[name="end"]::slotted(*) { opacity: 0.3;
 --ni-nimble-icon-color: var(--ni-nimble-button-label-font-color);
 }
}

@layer top {
  @media (prefers-reduced-motion) {
  .control { transition-duration: 0s;
 }
}
  :host([content-hidden]) [part="end"] { display: none;
 }
}

@layer base-selected {
  .control[aria-pressed="true"] [part="start"], .control[aria-pressed="true"] [part="end"] { --ni-nimble-icon-color: var(--ni-nimble-button-label-font-color);
 }
  .control[aria-pressed="true"] { background-color: transparent;
 color: var(--ni-nimble-button-label-font-color);
 background-image: linear-gradient(
                var(--ni-nimble-fill-selected-color),
                var(--ni-nimble-fill-selected-color)
            );
 border-color: rgba(var(--ni-nimble-fill-selected-rgb-partial-color), 0.3);
 }
}

@layer disabled {
  :host([disabled]) .control[aria-pressed="true"] { background-image: linear-gradient(
                var(--ni-nimble-fill-selected-color),
                var(--ni-nimble-fill-selected-color)
            );
 border-color: rgba(var(--ni-nimble-fill-selected-rgb-partial-color), 0.3);
 }
}

@layer base {
  .control { border-color: rgba(var(--ni-nimble-action-rgb-partial-color), 0.3);
 }
}

@layer disabled {
  :host([disabled]) .control { border-color: rgba(var(--ni-nimble-border-rgb-partial-color), 0.3);
 }
}

@layer base {
  :host([appearance-variant="primary"]) .control { border-color: var(--ni-nimble-action-rgb-partial-color);
 }
  :host([appearance-variant="accent"]) [part="start"], :host([appearance-variant="accent"]) [part="end"] { --ni-nimble-icon-color: var(--ni-nimble-button-accent-outline-font-color);
 }
  :host([appearance-variant="accent"]) .control { border-color: var(--ni-nimble-button-border-accent-outline-color);
 color: var(--ni-nimble-button-accent-outline-font-color);
 }
}

@layer active {
  :host([appearance-variant="accent"]) .control:active [part="start"], :host([appearance-variant="accent"]) .control:active [part="end"] { --ni-nimble-icon-color: var(--ni-nimble-button-accent-outline-font-color);
 }
  :host([appearance-variant="accent"]) .control:active { color: var(--ni-nimble-button-accent-outline-font-color);
 }
}

@layer base-selected {
  :host([appearance-variant="accent"]) .control[aria-pressed="true"] [part="start"], :host([appearance-variant="accent"]) .control[aria-pressed="true"] [part="end"] { --ni-nimble-icon-color: var(--ni-nimble-button-accent-outline-font-color);
 }
  :host([appearance-variant="accent"]) .control[aria-pressed="true"] { color: var(--ni-nimble-button-accent-outline-font-color);
 }
}
```

So the source code order matches the dom order well which has been a goal. But the order of the `@layer` definitions was surprising to me. And that the toggle buttons while first were not overridden? Or maybe the first one always wins? I don't know how it works.

Instead of overriding the `@layer` definition I'd rather have one clearly defined in the scope. I'm okay either:
- Merging all the layers into the top one in the button so button has some unused layers
- Making separate `css` tags for each layer config (button vs button+toggle) and updating each user of the button styles to explicitly pull iin only one `@layer` config.

Reply:
It's actually pretty straightforward. As the second "NOTE" in [this section of the spec](https://drafts.csswg.org/css-cascade-5/#layer-empty) states, layer ordering is defined by first occurrence of the layer name. So we can always declare the layer order before any other imported content without worrying about the imported content affecting our specified order. The only caveat is that if the imported content declares any layers that we didn't, those will be ordered after (take precedence over) the layers we declared. So, supposing we introduce a new layer in the shared button CSS, and we didn't update the toggle-button `styles.ts`, that layer would be applied last, which is probably wrong. For this reason, your recommendation seems like a good idea. I'll just add the new layer to the shared button styles.